### ISSUE TITLE
Unhandled Exception: ... invalid name ... at LiteDB.BsonDocument.set_Item in 'mutable' field

### ISSUE THREAD

**Issue by srlopez** [Zaid-Ajaj/LiteDB.FSharp#27]
The `mutable` attribute avoids insert in LiteDB, or passing through Json's encoder, something happens that the name of the field is modified.
```
open System
open LiteDB
open LiteDB.FSharp

[<StructuredFormatDisplay("a{SizeGb}GB")>]
[<CLIMutable>]
type Disk = 
    { SizeGb : int }

[<StructuredFormatDisplay("Computer #{Id}: {Manufacturer}/{DiskCount}/{Disks}")>]
[<CLIMutable>]
type Computer =
    { Id: int
      mutable Manufacturer: string
      mutable Disks: Disk list }
    [<BsonIgnore>] 
    member __.DiskCount = __.Disks.Length  // property calculated

[<EntryPoint>]
let main argv =

    let myPc =
        { Id = 0
          Manufacturer = "Computers Inc."
          Disks =
            [ { SizeGb = 100 }
              { SizeGb = 250 }
              { SizeGb = 500 } ] }
    printfn "%A" myPc

    let mapper = FSharpBsonMapper()
    use db = new LiteDatabase("data.db", mapper)
    let computers = db.GetCollection<Computer>("computers")

    // All right , but if I remove the comment on next line I get the error
    // computers.Insert(myPc) |> ignore
    myPc.Manufacturer <- "Dell"
    printfn "%A" myPc
    ...
```
The field you are trying to modify `Manufacturer` seems to change its name `Manufacturer@'`
The full error is:
```
Unhandled Exception: System.ArgumentException: Field 'Manufacturer@' has an invalid name.
   at LiteDB.BsonDocument.set_Item(String name, BsonValue value)
   at LiteDB.JsonReader.ReadObject()
   at LiteDB.JsonReader.ReadValue(JsonToken token)
   at LiteDB.JsonSerializer.Deserialize(String json)
   at LiteDB.FSharp.Bson.serialize[t](t entity)
   at LiteDB.FSharp.FSharpBsonMapper.ToDocument[t](t entity)
   at LiteDB.LiteCollection`1.Insert(T document)
   at Program.main(String[] argv) in /app/Program.fs:line ..
```

Thanks

**Comment by humhei** [comment]
```mutable Manufacturer```
[dnspy](https://github.com/0xd4d/dnSpy): Decompiled c# codes
```fsharp
public string Manufacturer@;
```

Converter will try to serialize both property `Manufacturer` and  field `Manufacturer@`


```immutable Manufacturer```
[dnspy](https://github.com/0xd4d/dnSpy): Decompiled c# codes
```fsharp
internal string Manufacturer@;
```

Converter will only try to serialize both property `Manufacturer`

**Comment by srlopez** [comment]
@humhei 
How can I fix it? It's a problem of my code or it's an authentic issue?

**Comment by humhei** [comment]
In fsharp world 
mutable fields is not recommanded(Not thread safe)

You should create a new record instead of setting property mutable

### workaround
```fsharp
open System
open LiteDB
open LiteDB.FSharp

[<StructuredFormatDisplay("a{SizeGb}GB")>]
[<CLIMutable>]
type Disk = 
    { SizeGb : int }

[<StructuredFormatDisplay("Computer #{Id}: {Manufacturer}/{DiskCount}/{Disks}")>]
[<CLIMutable>]
type Computer =
    { Id: int
      Manufacturer: string
      Disks: Disk list }
    [<BsonIgnore>] 
    member __.DiskCount = __.Disks.Length  // property calculated

[<EntryPoint>]
let main argv =

    let myPc =
        { Id = 0
          Manufacturer = "Computers Inc."
          Disks =
            [ { SizeGb = 100 }
              { SizeGb = 250 }
              { SizeGb = 500 } ] }
    printfn "%A" myPc

    let mapper = FSharpBsonMapper()
    use db = new LiteDatabase("data.db", mapper)
    let computers = db.GetCollection<Computer>("computers")

    let myDellPc = 
            {myPc withManufacturer = "Dell" }
    computers.Insert(myDellPc) |> ignore
    
...
```

**Comment by Zaid-Ajaj** [comment]
Hmm even though it is not *recommended*, I don't see an issue in supporting `mutable` constructs in the library, I will work on it...

**Comment by srlopez** [comment]
> Hmm even though it is not _recommended_, I don't see an issue in supporting `mutable` constructs in the library, I will work on it...

And  [<BsonIgnore>] toó, please

**Comment by Zaid-Ajaj** [comment]
I am afraid that `[<BsonIgnore>]` doesn't really make sense within F# records. Using a member within the record is the way to go as it will be used to compute properties derived from the record fields and doesn't get persisted (= ignored) as follows:
```fs
[<CLIMutable>]
type Computer =
    { Id: int
      Manufacturer: string
      Disks: Disk list }

    // this is not persisted
    with member this.Ignored = sprintf "%d => %s" this.Id this.Manufacturer
```

Aren't these members good enough for your use-case? 

> we can continue discussion at #26

As for `[<BsonField>]` I think it is possible but will require more work, I will open another issue for it.

**Comment by Zaid-Ajaj** [comment]
Fixed and released in version `2.8.0` to nuget, should be available really soon when nuget does it's magic. Now I decided to take control of how the fields are marshalled, so now these mutable records will work just fine :smile:
